firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/FeedbackTriggers.java

```diff
@@ -0,0 +1,138 @@
+import static android.content.pm.PackageManager.PERMISSION_GRANTED;
+
+import android.app.AlertDialog;
+import android.content.Context;
+import android.database.ContentObserver;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Handler;
+import android.provider.MediaStore;
+import android.provider.MediaStore.Images.Media;
+import android.util.Log;
+import androidx.activity.ComponentActivity;
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.core.content.ContextCompat;
+import com.google.firebase.appdistribution.FirebaseAppDistribution;
+import java.util.HashSet;
+import java.util.Set;
+
+class FeedbackTriggers extends ContentObserver {
+
+  private static final String TAG = "FeedbackTriggers";
+  private static final boolean SHOULD_CHECK_IF_PENDING = Build.VERSION.SDK_INT == 29;
+  private static final String[] PROJECTION =
+      SHOULD_CHECK_IF_PENDING
+          ? new String[] {MediaStore.Images.Media.DATA}
+          : new String[] {
+            MediaStore.Images.Media.DATA, android.provider.MediaStore.MediaColumns.IS_PENDING
+          };
+  private final Set<String> seenImages = new HashSet<>();
+
+  private final Context context;
+  private final CharSequence infoText;
+
+  private final ActivityResultLauncher<String> requestPermissionLauncher;
+
+  private Uri currentUri;
+
+  /**
+   * Creates a FeedbackTriggers instance for an activity.
+   *
+   * <p>This must be called during activity initialization.
+   *
+   * @param activity The host activity
+   * @param handler The handler to run {@link #onChange} on, or null if none.
+   */
+  public FeedbackTriggers(ComponentActivity activity, CharSequence infoText, Handler handler) {
+    super(handler);
+    this.context = activity;
+    this.infoText = infoText;
+
+    // Register the permissions launcher
+    requestPermissionLauncher =
+        activity.registerForActivityResult(
+            new ActivityResultContracts.RequestPermission(),
+            isGranted -> {
+              if (isGranted) {
+                maybeStartFeedbackForScreenshot(currentUri);
+              } else {
+                new AlertDialog.Builder(context)
+                    .setMessage(
+                        "Without storage access, screenshots taken while in the app will not be able to automatically start the feedback flow. To enable this feature, grant the app storage access in Settings.")
+                    .setPositiveButton(
+                        "OK",
+                        (a, b) -> {
+                          /* do nothing */
+                        })
+                    .show();
+              }
+            });
+  }
+
+  void registerScreenshotObserver() {
+    context
+        .getContentResolver()
+        .registerContentObserver(Media.EXTERNAL_CONTENT_URI, true /*notifyForDescendants*/, this);
```

Comment:
supernit: I'm partial to the comment coming first with a `=`: `/* notifyForDescendants= */ true`